wlp passwords keys may fail to decode

**Describe the bug**  
Certain values for `wlp.password.encryption.key` and `wlp.aes.encryption.key` fail to decode passwords. 
Example `wlp.aes.encryption.key` value: `3ORhx1L0ME//P2JDl1elDjOqhhagCoMAZ4XFbhQxJoM=`
```
java.security.spec.InvalidKeySpecException: The value that is specified in the wlp.aes.encryption.key variable must be base64 encoded.
	at com.ibm.ws.crypto.util.AESKeyManager$KeyVersion.decodeAesBase64Key(AESKeyManager.java:110)
	at com.ibm.ws.crypto.util.AESKeyManager$KeyVersion.get(AESKeyManager.java:84)
	at com.ibm.ws.crypto.util.AESKeyManager$KeyVersion.access$000(AESKeyManager.java:43)
	at com.ibm.ws.crypto.util.AESKeyManager.getHolder(AESKeyManager.java:172)
	at com.ibm.ws.crypto.util.AESKeyManager.getKey(AESKeyManager.java:158)
	at com.ibm.ws.crypto.util.PasswordCipherUtil.aesDecipherCommon(PasswordCipherUtil.java:309)
	at com.ibm.ws.crypto.util.PasswordCipherUtil.aesDecipherV2(PasswordCipherUtil.java:686)
	at com.ibm.ws.crypto.util.PasswordCipherUtil.aesDecipher(PasswordCipherUtil.java:281)
	at com.ibm.ws.crypto.util.PasswordCipherUtil.decipher(PasswordCipherUtil.java:219)
	at com.ibm.websphere.crypto.PasswordUtil.decode_password(PasswordUtil.java:649)
	at com.ibm.websphere.crypto.PasswordUtil.passwordDecode(PasswordUtil.java:448)
	at com.ibm.ws.security.auth.data.internal.AuthDataImpl.lambda$activate$0(AuthDataImpl.java:55)
	at io.openliberty.checkpoint.spi.CheckpointPhase.onRestore(CheckpointPhase.java:377)
	at io.openliberty.checkpoint.spi.CheckpointPhase.onRestore(CheckpointPhase.java:358)
	at com.ibm.ws.security.auth.data.internal.AuthDataImpl.activate(AuthDataImpl.java:54)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:586)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.invokeMethod(BaseMethod.java:257)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.access$500(BaseMethod.java:41)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod$Resolved.invoke(BaseMethod.java:701)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.invoke(BaseMethod.java:544)
	at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:317)
	at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:307)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.createImplementationObject(SingleComponentManager.java:354)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.createComponent(SingleComponentManager.java:115)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.getService(SingleComponentManager.java:1002)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.getServiceInternal(SingleComponentManager.java:975)
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:776)
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.enableInternal(AbstractComponentManager.java:674)
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.enable(AbstractComponentManager.java:437)
	at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.enableComponents(ConfigurableComponentHolder.java:671)
	at org.apache.felix.scr.impl.BundleComponentActivator.initialEnable(BundleComponentActivator.java:310)
	at org.apache.felix.scr.impl.Activator.loadComponents(Activator.java:613)
	at org.apache.felix.scr.impl.Activator.access$200(Activator.java:75)
	at org.apache.felix.scr.impl.Activator$ScrExtension.start(Activator.java:480)
	at org.apache.felix.scr.impl.AbstractExtender.createExtension(AbstractExtender.java:196)
	at org.apache.felix.scr.impl.AbstractExtender.modifiedBundle(AbstractExtender.java:169)
	at org.apache.felix.scr.impl.AbstractExtender.modifiedBundle(AbstractExtender.java:49)
	at org.osgi.util.tracker.BundleTracker$Tracked.customizerModified(BundleTracker.java:490)
	at org.osgi.util.tracker.BundleTracker$Tracked.customizerModified(BundleTracker.java:1)
	at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:233)
	at org.osgi.util.tracker.BundleTracker$Tracked.bundleChanged(BundleTracker.java:452)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:975)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:151)
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEventPrivileged(EquinoxEventPublisher.java:237)
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:136)
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:128)
	at org.eclipse.osgi.internal.framework.EquinoxContainerAdaptor.publishModuleEvent(EquinoxContainerAdaptor.java:232)
	at org.eclipse.osgi.container.Module.publishEvent(Module.java:534)
	at org.eclipse.osgi.container.Module.start(Module.java:518)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel$2.run(ModuleContainer.java:2110)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1595)
Caused by: java.lang.IllegalArgumentException: Input byte array has wrong 4-byte ending unit
	at java.base/java.util.Base64$Decoder.decode0(Base64.java:842)
	at java.base/java.util.Base64$Decoder.decode(Base64.java:570)
	at java.base/java.util.Base64$Decoder.decode(Base64.java:593)
	at com.ibm.ws.crypto.util.AESKeyManager$KeyVersion.decodeAesBase64Key(AESKeyManager.java:108)
	... 54 more
```

**Steps to Reproduce**  
Create a wlp.password.encryption.key or wlp.aes.encryption.key with `//` in the value
encode a password
add the password to server.xml 
on server start you'll see the base64 failure if you used byok or you will see a AEAD 

**Expected behavior**  
 
they key should be loaded and password should be decoded. 

**Diagnostic information:**  
 - OpenLiberty Version: 25.0.0.12 for BYOK and likely since release for pre-BYOK aes passwords 
 
 - server.xml configuration (WITHOUT sensitive information like passwords)

Symptom may also show as
```
CWWKS1862E: AES password decryption failed. Check your java.security settings and add the missing algorithm.
java.security.spec.InvalidKeySpecException: The value that is specified in the wlp.aes.encryption.key variable must be base64 encoded.
    at com.ibm.ws.crypto.util.AESKeyManager$KeyVersion.decodeAesBase64Key(AESKeyManager.java:110)
    at com.ibm.ws.crypto.util.AESKeyManager$KeyVersion.get(AESKeyManager.java:84)
    at com.ibm.ws.crypto.util.AESKeyManager$KeyVersion.access$000(AESKeyManager.java:43)
    at com.ibm.ws.crypto.util.AESKeyManager.getHolder(AESKeyManager.java:172)
    at com.ibm.ws.crypto.util.AESKeyManager.getKey(AESKeyManager.java:158)
    at com.ibm.ws.crypto.util.PasswordCipherUtil.aesDecipherCommon(PasswordCipherUtil.java:309)
    at com.ibm.ws.crypto.util.PasswordCipherUtil.aesDecipherV2(PasswordCipherUtil.java:686)
    at com.ibm.ws.crypto.util.PasswordCipherUtil.aesDecipher(PasswordCipherUtil.java:281)
    at com.ibm.ws.crypto.util.PasswordCipherUtil.decipher(PasswordCipherUtil.java:219)
    at com.ibm.websphere.crypto.PasswordUtil.decode_password(PasswordUtil.java:649)
    at com.ibm.websphere.crypto.PasswordUtil.passwordDecode(PasswordUtil.java:448)
    at com.ibm.ws.security.token.ltpa.internal.LTPAKeyCreateTask.getKeyPasswordBytes(LTPAKeyCreateTask.java:52)
    at com.ibm.ws.security.token.ltpa.internal.LTPAKeyCreateTask.getPreparedLtpaKeyInfoManager(LTPAKeyCreateTask.java:59)
    at com.ibm.ws.security.token.ltpa.internal.LTPAKeyCreateTask.createRequiredCollaborators(LTPAKeyCreateTask.java:95)
    at com.ibm.ws.security.token.ltpa.internal.LTPAKeyCreateTask.run(LTPAKeyCreateTask.java:105)
    at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:344)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
    at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.IllegalArgumentException: Last unit does not have enough valid bits
    at java.base/java.util.Base64$Decoder.decode0(Base64.java:868)
    at java.base/java.util.Base64$Decoder.decode(Base64.java:566)
    at java.base/java.util.Base64$Decoder.decode(Base64.java:589)
    at com.ibm.ws.crypto.util.AESKeyManager$KeyVersion.decodeAesBase64Key(AESKeyManager.java:108)
```

--- a/dev/com.ibm.ws.crypto.passwordutil/src/com/ibm/ws/crypto/util/VariableResolver.java
+++ b/dev/com.ibm.ws.crypto.passwordutil/src/com/ibm/ws/crypto/util/VariableResolver.java
@@ -1,10 +1,10 @@
 /*******************************************************************************
- * Copyright (c) 2012 IBM Corporation and others.
+ * Copyright (c) 2012, 2025 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License 2.0
  * which accompanies this distribution, and is available at
  * http://www.eclipse.org/legal/epl-2.0/
- * 
+ *
  * SPDX-License-Identifier: EPL-2.0
  *
  * Contributors:
@@ -32,6 +32,6 @@ public class VariableResolver implements KeyStringResolver {
 
     @Override
     public char[] getKey(String val) {
-        return registry.resolveString(val).toCharArray();
+        return registry.resolveRawString(val).toCharArray();
     }
 }

--- a/dev/com.ibm.ws.crypto.passwordutil/src/com/ibm/ws/crypto/util/VariableResolver.java
+++ b/dev/com.ibm.ws.crypto.passwordutil/src/com/ibm/ws/crypto/util/VariableResolver.java
@@ -32,6 +32,7 @@ public class VariableResolver implements KeyStringResolver {
 
     @Override
     public char[] getKey(String val) {
+        // Use resolveRawString instead of resolveString so that VariableRegistry doesn't normalize the value saved in server config.
         return registry.resolveRawString(val).toCharArray();
     }
 }

